pip/pep425tags.py

```diff
@@ -122,6 +122,42 @@ def get_platform():
     return distutils.util.get_platform().replace('.', '_').replace('-', '_')
 
 
+def get_darwin_arches(major, minor, machine):
+    """Return a list of supported arches (including group arches) for
+    the given major, minor and machine architecture of an OS X machine.
+    """
+    arches = []
+
+    def _supports_arch(major, minor, arch):
+        if arch == 'ppc':
+            return (major, minor) <= (10, 5)
+        if arch == 'ppc64':
+            return (major, minor) >= (10, 2) and (major, minor) <= (10, 5)
+        if arch == 'i386':
+            return (major, minor) >= (10, 4) and (major, minor) <= (10, 6)
+        if arch == 'x86_64':
+            return (major, minor) >= (10, 4)
+        if arch in groups:
+            for garch in groups[arch]:
+                if _supports_arch((major, minor), garch):
+                    return True
+        return False
+
+    groups = {'fat': ('i386', 'ppc'),
+              'intel': ('x86_64', 'i386'),
+              'fat64': ('x86_64', 'ppc64'),
+              'fat32': ('x86_64', 'i386', 'ppc')
+              }
+    if _supports_arch(major, minor, machine):
+        arches.append(machine)
+    for garch in groups:
+        if machine in garch and _supports_arch(major, minor, garch):
```

Comment:
That is what I want; but the `_supports_arch` call actually is correct, because (for example) `macosx_10_0` doesn't support `i386`, but does support `ppc`, which is in `fat32`, which provides `i386`. So `macosx_10_0_i386` is not a valid tag but `macosx_10_0_fat32` is.